### PR TITLE
fix: update actions to remove deprecated nodejs versions

### PR DESCRIPTION
**Pull request by bruuuuuuuce**
There are a ton of deprecation warnings when running github actions within this repo

image

This pr takes a stab at trying to eliminate some of them

### CODE REVIEW

File: .github/workflows/on-pull-request.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     name: Style & Lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (bruuuuuuuce) [inline .github/workflows/on-pull-request.yml:16]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/on-pull-request.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
       max-parallel: 1
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (bruuuuuuuce) [inline .github/workflows/on-pull-request.yml:34]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/on-pull-request.yml [hunk 3]
```diff
@@ -53,7 +53,7 @@ jobs:
   build-windows:
     runs-on: windows-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (bruuuuuuuce) [inline .github/workflows/on-pull-request.yml:56]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/on-push-to-main.yml [hunk 4]
```diff
@@ -29,7 +29,7 @@ jobs:
       max-parallel: 1
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (bruuuuuuuce) [inline .github/workflows/on-push-to-main.yml:32]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/on-push-to-main.yml [hunk 5]
```diff
@@ -48,7 +48,7 @@ jobs:
   build-windows:
     runs-on: windows-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (bruuuuuuuce) [inline .github/workflows/on-push-to-main.yml:51]:
```suggestion
      - uses: actions/checkout@v4
```

**Review by cprice404 — APPROVED** [review]
ty

**Pull request merged**